R/nnmf.R

```diff
@@ -0,0 +1,131 @@
+#' Non-Negative Matrix Factorization
+#'
+#' S4 Class implementing NNMF.
+#'
+#' NNMF is a method for decomposing a matrix into a smaller 
+#'  dimension such that the constraint that the data (and the 
+#'  projection) are not negative is taken into account.
+#'
+#' @template dimRedMethodSlots
+#'
+#' @template dimRedMethodGeneralUsage
+#'
+#' @section Parameters:
+#' The method can take the following parameters:
+#' \describe{
+#'   \item{ndim}{The number of output dimensions.}
+#'   \item{method}{character, should be either "scd" for (sequential 
+#'    coordinate-wise descent) or "lee" (for Lee's multiplicative algorithm).}
+#'   \item{loss}{character, should be either  "mse" for mean square error 
+#'    or "mkl" for mean Kullback–Leibler-divergence.}
+#'   \item{max.iter}{integer, maximum number of iterations (default: 500).}
+#'   \item{n.threads}{integer, number of threads (default: no parallelism)}
+#'   \item{verbose}{integer between 0 and 2, for amount of logging. Defaults
+#'    to 0L.}
+#'   \item{rel.tol}{numeric, convergence criterion that defaults to 1e-04.}
+#' }
+#'
+#' @section Implementation:
+#'
+#' Wraps around \code{\link[NNLM]{nnmf}}. Note that the estimation uses random
+#'  numbers. To create reproducible results, set the random number seed prior
+#'  to execution. 
+#'
+#' @examples
+#' dat <- loadDataSet("Iris")
+#' 
+#' set.seed(4646)
+#' factorization <- embed(dat, "NNMF")
+#' 
+#' proj_dat <- factorization@apply(dat)
+#'
+#' plot(proj_dat@data[, 1], proj_dat@data[, 2])
+#'
+#' @include dimRedResult-class.R
+#' @include dimRedMethod-class.R
+#' @family dimensionality reduction methods
+#' @export NNMF
+#' @exportClass NNMF
+#' @importFrom NNLM nnmf nnlm
```

Comment:
I put all the packages `dimRed` depends on in the `Suggests` section of the `DESCRIPTION` file and I don't import anything directly if I can avoid it. There is `chckpkg` to check for dependencies. 
`dimRed` depends on many packages and this way I avoid problems with CRAN if one of these packages should ever get removed from CRAN.

Adding NNLM to DESCRIPTION/Suggests and removing this line should also get you through the first set of `R CMD check` tests on travis.